### PR TITLE
Fixing redundant ignored methods in JUnit Reports

### LINKED ISSUE #1265: [Bug] JUnit Reporter includes redundant ignored methods
### TestNG Version
6.10

### Expected behavior

In the JUnit xml report files I should see entries `<skipped/>` only for tests which have been marked with `@Test(enabled = false)`.

### Actual behavior

Now, in  JUnit xml report files I can see a plethora redundant `<skipped/>` entries. Each test report contain literally all methods which have been inherited from the parent test, executed once, and thus skipped by the TestNG for remaining inherited classes.

Whereas I believe that only methods annotated with a `@Test(enabled = false)` should appear in the JUnit report.

EDIT: it is even worse: in each and every report, there are entries for methods marked with `@Test(enabled = false)` which come from totally different classes and have nothing to do with the actual class (look at the `should_be_ignored` method in the example below).

### Is the issue reproductible on runner?

- [x ] Maven

### Test case sample

```java
import org.testng.annotations.AfterSuite;
import org.testng.annotations.BeforeSuite;

public abstract class ParentTest {

    @BeforeSuite
    public synchronized void startEverything() throws Exception {
    }

    @AfterSuite
    public synchronized void shutdownContainer() throws Exception {
    }
}
```
FirstTest.java:

```java
import static org.testng.Assert.*;
import org.testng.annotations.Test;

public class FirstTest extends ParentTest {

    @Test
    public void should_pass() {
        assertEquals("abc", "abc");
    }

    @Test(enabled = false)
    public void should_be_ignored() {
        assertEquals("abcd", "abc");
    }
}
```

SecondTest.java:

```java
public class SecondTest extends ParentTest {

    @Test
    public void should_pass_second() {
        assertEquals("abc", "abc");
    }
}
```

ThirdTest.java:

```java
public class ThirdTest extends ParentTest {

    @Test
    public void should_pass_third() {
        assertEquals("abc", "abc");
    }
}
```

JUnit output is extremely weird:

junitreports/TEST-abc.FirstTest.xml:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite name="abc.FirstTest" tests="1" time="0.001" ignored="5" failures="0" timestamp="08 Dec 2016 10:53:32 GMT" skipped="0" hostname="EPPLGDAW0188" errors="0">
  <testcase name="should_pass" time="0.001" classname="abc.FirstTest"/>
  <testcase name="shutdownContainer" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- shutdownContainer -->
  <testcase name="shutdownContainer" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- shutdownContainer -->
  <testcase name="startEverything" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- startEverything -->
  <testcase name="startEverything" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- startEverything -->
  <testcase name="should_be_ignored" classname="abc.FirstTest">
    <skipped/>
  </testcase> <!-- should_be_ignored -->
</testsuite> <!-- abc.FirstTest -->
```

junitreports/TEST-abc.SecondTest.xml:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite name="abc.SecondTest" tests="1" time="0.005" ignored="5" failures="0" timestamp="08 Dec 2016 10:53:32 GMT" skipped="0" hostname="EPPLGDAW0188" errors="0">
  <testcase name="should_pass_second" time="0.005" classname="abc.SecondTest"/>
  <testcase name="shutdownContainer" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- shutdownContainer -->
  <testcase name="shutdownContainer" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- shutdownContainer -->
  <testcase name="startEverything" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- startEverything -->
  <testcase name="startEverything" classname="abc.ParentTest">
    <skipped/>
  </testcase> <!-- startEverything -->
  <testcase name="should_be_ignored" classname="abc.FirstTest">
    <skipped/>
  </testcase> <!-- should_be_ignored -->
</testsuite> <!-- abc.SecondTest -->
```

`junitreports/TEST-abc.ThirdTest.xml ` is the same, to I skip it here.

It seems to me that this whole mess is a result of implementing the issue #1211 .

Kind regards,
Greg

### PR DESCRIPTION
**Pull request by krmahadevan**
Fixes # .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

Fixes #1265

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @juherr

File: src/main/java/org/testng/reporters/JUnitReportReporter.java [hunk 1]
```diff
@@ -172,9 +169,16 @@ public void generateReport(List<XmlSuite> xmlSuites, List<ISuite> suites,
...
 
-//    System.out.println(xsb.toXML());
-//    System.out.println("");
+  }
 
+  private static int getDisabledTestCount(Set<ITestNGMethod> methods) {
+    int count = 0;
+    for (ITestNGMethod method : methods) {
+      if (!method.getEnabled()) {
+        count = count + 1;
```
Comment (juherr) [inline src/main/java/org/testng/reporters/JUnitReportReporter.java]: could be `count++`

File: src/test/java/test/SimpleBaseTest.java [hunk 2]
```diff
@@ -70,13 +70,22 @@ protected static TestNG create(Path outputDir, List<XmlSuite> suites) {
...
     int i=0;
     for (Class<?> testClass : testClasses) {
       createXmlTest(suite, testClass.getName() + i, testClass);
       i++;
     }
-    return create(suite);
+    if (outDir == null) {
+      return create(suite);
+    }
+    return create(outDir, suite);
```
Comment (juherr) [inline src/test/java/test/SimpleBaseTest.java]: `create(outDir, suite)` should manage `null`

**Review by juherr — APPROVED** [review]
Few open points but LGTM. Ping @cbeust :+1:

**Comment by krmahadevan** [comment]
@cbeust - Can you please help get this merged. Looks like the fix for #1262 is in the same set of files. So it would be good if this PR can be merged, so that I can proceed further with #1262

**Comment by krmahadevan** [comment]
ping @cbeust ... I can raise my next PR only after this change is merged.. pls help merge :)

**Pull request merged**